hail/python/hailtop/aiocloud/aioaws/fs.py

```diff
@@ -281,13 +281,16 @@ async def open(self, url: str) -> ReadableStream:
         except self._s3.exceptions.NoSuchKey as e:
             raise FileNotFoundError(url) from e
 
-    async def open_from(self, url: str, start: int) -> ReadableStream:
+    async def open_from(self, url: str, start: int, *, length: Optional[int] = None) -> ReadableStream:
         bucket, name = self._get_bucket_name(url)
+        range_str = f'bytes={start}-'
+        if length is not None:
+            range_str += str(start + length)
```

Comment:
Can we double check the definitions of the range are correct for all clouds? I vaguely recall Google might have different semantics and the end point is inclusive. 

https://cloud.google.com/storage/docs/samples/storage-download-byte-range#storage_download_byte_range-python

Reply:
Yeah, you're totally right it's inclusive. I'll fix.
- https://cloud.google.com/storage/docs/xml-api/reference-headers#range
- https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Range
- https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObject.html#API_GetObject_RequestSyntax
- https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.35